scripts/gen.py

```diff
@@ -507,8 +520,9 @@ def extract_list(t, l):
 
 def tuple_type_list(t):
   assert isinstance(t, lark.tree.Tree)
-  c = t.children[0]
-  assert isinstance(c, lark.tree.Tree) and c.data == 'core_type'
+  for c in t.children:
```

Comment:
This looks wrong, especially when below we fetch `children[0]`.
Fetching the type is only allowed when a tuple has all the same element types.

Reply:
Hmmm I wanted to extend this function to skip the `const` (as shown in #2) if exists, that's why I used a break here. Do you have a better idea how to extract Tensor out of both cases? Thanks!
```
#1 Tree(type, [Tree(core_type, [Tree(template, [Token(TNAME, 'c10::optional'), Tree(typelist, [Tree(type, [Tree(core_type, [Token(TNAME, 'int64_t')])])])])])])
#2 Tree(type, [Token(CONST, 'const'), Tree(core_type, [Tree(template, [Token(TNAME, 'c10::optional'), Tree(typelist, [Tree(type, [Tree(core_type, [Token(TNAME, 'Tensor')])])])])]), Tree(refspec, [Token(REF, '&')])])
```

Reply:
I would not make tuples hold references.
They could, but they can be error prone (tuples going around when reference sources are gone).